lib/_http_outgoing.js

```diff
@@ -407,9 +424,10 @@ Object.defineProperty(OutgoingMessage.prototype, 'headersSent', {
 
 OutgoingMessage.prototype.write = function(chunk, encoding, callback) {
   var self = this;
+  var err;
 
   if (this.finished) {
-    var err = new Error('write after end');
+    err = new Error('write after end');
```

Comment:
It would be slightly more memory-efficient here and elsewhere to create the Error object inside the nextTick callback.